protocol/contracts/beanstalk/sun/SeasonFacet/Distribution.sol

```diff
@@ -0,0 +1,154 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.20;
+
+import {C} from "contracts/C.sol";
+import {SafeCast} from "@openzeppelin/contracts/utils/math/SafeCast.sol";
+import {AppStorage, Storage} from "contracts/beanstalk/AppStorage.sol";
+import {ShipmentPlan, Receiving} from "./Receiving.sol";
+
+/**
+ * @title Distribution
+ * @author funderbrker
+ * @notice Handles shipping of new Bean mints.
+ */
+contract Distribution is Receiving {
+    using SafeCast for uint256;
+
+    /**
+     * @notice Emitted during Sunrise when Beans are distributed to the Field, the Silo, and Fertilizer.
+     * @param season The Season in which Beans were distributed.
+     * @param shipmentRoutes The routes defining where to distribute.
+     * @param shipmentAmounts The amount of Beans distributed to each route.
+     */
+    event Ship(
+        uint32 indexed season,
+        Storage.ShipmentRoute[] shipmentRoutes,
+        uint256[] shipmentAmounts
+    );
+
+    //////////////////// REWARD BEANS ////////////////////
+
+    /**
+     * @notice Mints and distributes Beans across all active shipping routes.
+     * @param beansToShip The total number of Beans to distribute.
+     */
+    function ship(uint256 beansToShip) internal {
+        C.bean().mint(address(this), beansToShip);
+
+        Storage.ShipmentRoute[] memory shipmentRoutes = s.shipmentRoutes;
+        ShipmentPlan[] memory shipmentPlans = new ShipmentPlan[](shipmentRoutes.length);
+        uint256[] memory shipmentAmounts = new uint256[](shipmentRoutes.length);
+        uint256 totalPoints;
+        (shipmentPlans, totalPoints) = getShipmentPlans(shipmentRoutes);
+
+        // May need to calculate individual stream rewards multiple times, since
+        // they are dependent on each other and each has a cap.
+        for (uint256 i; i < shipmentRoutes.length; i++) {
+            bool capExceeded;
+            // Calculate the amount of rewards to each stream. Ignores cap and plans with 0 points.
+            getBeansFromPoints(shipmentAmounts, shipmentPlans, totalPoints, beansToShip);
```

Comment:
is `shipmentAmounts` passed by reference or pass by value? From my understanding it was passed by value.

Reply:
non-value types (like arrays) are always passed by reference

Reply:
dope, ty